datafusion-federation/src/sql/mod.rs

```diff
@@ -100,6 +113,481 @@ impl OptimizerRule for SQLFederationOptimizerRule {
         false
     }
 }
+
+/// Rewrite table scans to use the original federated table name.
+fn rewrite_table_scans(
```

Comment:
This is needed to support the table name in the remote source being different from the table name registered in DataFusion. This required a lot of trial and error to get right, and we have a good integration test coverage of this in spiceai.